test/unit/package_manager_apt_test.py

```diff
@@ -0,0 +1,170 @@
+
+    @patch('kiwi.logger.log.warning')
+    def test_request_collection(self, mock_log_warn):
+        self.manager.request_collection('name')
+        assert self.manager.collection_requests == []
+        assert mock_log_warn.called
+
+    @patch('kiwi.logger.log.warning')
+    def test_request_product(self, mock_log_warn):
+        self.manager.request_product('name')
+        assert self.manager.product_requests == []
+        assert mock_log_warn.called
+
+    @raises(KiwiRuntimeError)
+    def test_process_install_requests_bootstrap_no_dist(self):
+        self.manager.distribution = None
+        self.manager.process_install_requests_bootstrap()
+
+    @patch('os.path.exists')
+    @raises(KiwiRequestError)
+    def test_process_install_requests_bootstrap_no_debootstrap_script(
+        self, mock_exists
+    ):
+        mock_exists.return_value = False
+        self.manager.process_install_requests_bootstrap()
+
+    @patch('kiwi.command.Command.run')
+    @patch('os.path.exists')
+    @patch('kiwi.package_manager.apt.Path.wipe')
+    @raises(KiwiRequestError)
+    def test_process_install_requests_bootstrap_failed_debootstrap(
+        self, mock_wipe, mock_exists, mock_run
+    ):
+        self.manager.request_package('apt-get')
+        mock_run.side_effect = Exception
+        mock_exists.return_value = True
+        self.manager.process_install_requests_bootstrap()
+
+    @patch('kiwi.command.Command.call')
+    @patch('kiwi.command.Command.run')
+    @patch('os.path.exists')
+    @patch('kiwi.package_manager.apt.DataSync')
+    def test_process_install_requests_bootstrap(
+        self, mock_sync, mock_exists, mock_run, mock_call
+    ):
+        self.manager.request_package('apt-get')
+        self.manager.request_package('vim')
+        data = mock.Mock()
+        mock_sync.return_value = data
+        mock_exists.return_value = True
+        self.manager.process_install_requests_bootstrap()
+        mock_sync.assert_called_once_with(
+            'root-dir.debootstrap/', 'root-dir'
+        )
+        data.sync_data.assert_called_once_with(
+            options=['-a', '-H', '-X', '-A']
+        )
+        assert mock_run.call_args_list == [
+            call(command=['mountpoint', 'root-dir/dev'], raise_on_error=False),
+            call(
+                [
+                    'debootstrap', '--no-check-gpg', 'xenial',
+                    'root-dir.debootstrap', 'xenial_path'
+                ], ['env']
+            ),
+            call(
+                [
+                    'bash', '-c',
+                    'rm -r -f root-dir.debootstrap &&' + \
+                    ' chroot root-dir apt-get root-moved-arguments update'
+                ], ['env']
+            )
+        ]
+        mock_call.assert_called_once_with(
+            [
+                'chroot', 'root-dir', 'apt-get',
+                'root-moved-arguments', 'install', 'vim'
+            ], ['env']
+        )
+
```

Comment:
You can contract all of that above to a bit more sparse code. :wink: Up to you though, but I would prefer less scrolling over ```[
     [ 
               {
        [
                      (
           [
                        (
               [
                   (
                              (
           [
                    (
                       )
              ]
               ,
           ]
      )
                  ]
         }
     ]
```
:wink: